utils/R/convert.input.R

```diff
@@ -6,65 +6,144 @@
+                  "so that existing input can be updated while maintaining continuous time span."))
+      
+      # There might be existing files for some years (but not all; checked that above)
+      # fcn should be smart enough not overwrite the existing ones, 
+      # and hopefully won't waste time working on them either
+      # At the end, if convert.inputs was successful we'll need to update its start/end dates
+      # We don't know the dbfile path/prefix until after fcn runs, so unfortunately can't check 
+      # that the new dbfile record won't conflict with existing ones. 
+    }
+  } else {
+    # No existing record found. Should be good to go?
   }
   
-  input = db.query(paste("SELECT * from inputs where id =",input.id),con)
-  if(nrow(input)==0){logger.error("input not found",input.id);return(NULL)}
   
-  ifelse(host$name == "localhost", machine.host <- fqdn(), machine.host <- host$name)
-  machine = db.query(paste0("SELECT * from machines where hostname = '",machine.host,"'"),con)
-  # machine = db.query(paste("SELECT * from machines where id = ",dbfile$machine_id),con)
-  if(nrow(machine)==0){logger.error("machine not found",host$name);return(NULL)}
   
-  # dbfile may return more than one row -> may need to loop over machine ids
-  dbfile = db.query(paste("SELECT * from dbfiles where container_id =",input.id," and container_type = 'Input' and machine_id =",machine$id),con)
-  if(nrow(dbfile)==0){logger.error("dbfile not found",input.id);return(NULL)}
-  if(nrow(dbfile)>1){
-    logger.warning("multiple dbfile records, using last",dbfile);
-    dbfile = dbfile[nrow(dbfile),]
+  machine.host <- ifelse(host$name == "localhost", fqdn(), host$name)
+  machine = db.query(paste0("SELECT * from machines where hostname = '", machine.host, "'"), con)
+  
+  if(nrow(machine)==0){
+    logger.error("machine not found", host$name)
+    return(NULL)
   }
 
+  if(missing(input.id) || is.na(input.id) || is.null(input.id)) {
+    input <- dbfile <- NULL
+  } else {
+    input = db.query(paste("SELECT * from inputs where id =", input.id), con)
+    if(nrow(input)==0){
+      logger.error("input not found", input.id)
+      return(NULL)
+    }
+    
+    dbfile = db.query(paste(
+      "SELECT * from dbfiles where container_id =", input.id, 
+      " and container_type = 'Input' and machine_id =", machine$id), con)
+    if(nrow(dbfile)==0) {
+      logger.error("dbfile not found",input.id);return(NULL)
+    }
+    if(nrow(dbfile)>1) {
+      logger.warning("multiple dbfile records, using last",dbfile);
+      dbfile = dbfile[nrow(dbfile),]
+    }
+  } 
+
   #--------------------------------------------------------------------------------------------------#
   # Perform Conversion 
-  
   conversion = "local.remote" #default
   
-  if(!is.null(browndog) & host$name == 'localhost'){ # perform conversions with Brown Dog - only works locally right now
+  if(!is.null(browndog) && host$name == 'localhost'){ 
```

Comment:
To reiterate, I did *nothing* functional to code within browndog-only blocks. Apparently I obsessive-compulsively did some formatting, though.